### ISSUE TITLE
unclear instructions for free'ing a wants_verifier

### ISSUE THREAD

**Issue by kevinburke** [rustls/rustls-ffi#162]
The documentation for `rustls_client_config_builder_wants_verifier_free` reads:

```c
/**
 * "Free" a client_config_builder_wants_verifier before transmogrifying it into a client_config.
 * Normally builders are consumed to client_configs via `rustls_client_config_builder_build`
 * and may not be free'd or otherwise used afterwards.
 * Use free only when the building of a config has to be aborted before a config
 * was created.
 */
void rustls_client_config_builder_wants_verifier_free(struct rustls_client_config_builder_wants_verifier *builder);
```

However, you do not pass a `rustls_client_config_builder_wants_verifier` to `rustls_client_config_builder_build`, you pass a `rustls_client_config_builder`. So I am unsure of if or when a `wants_verifier` must be free'd.

It's possible that the `wants_verifier` gets consumed while running `rustls_client_config_builder_dangerous_set_certificate_verifier`, but the documentation does not make that clear.

I tried checking to see what `tests/client.c` does to see if I could get a hint. It looks like there are two code paths where we init a `wants_verifier` and a `config_builder`, but don't end up calling `rustls_client_config_builder_build`:

- if the `load_roots_from_file` call fails to return RUSTLS_RESULT_OK
- if neither CA_FILE nor NO_CHECK_CERTIFICATE environment variables are set.

I would be happy to update the tests but I'm not sure where or if the wants_verifier should be free'd.

**Comment by kevinburke** [comment]
Maybe this will all get sorted by #154 and I should just wait. I do think the tests need at least one call to free() though.

**Comment by jsha** [comment]
Resolved by #154
